### PR TITLE
fix: import compilation for older solidity compilers

### PR DESCRIPTION
**Pull request by folkyatina**
### What I did

fixes: #65 

### How I did it
Added a test for the case. 
And in the code, appended a base_path to each relative path in remapping to make it absolute back if compiler does not support `--base_path` flag. 

I'm a very inexperienced pythonist, so I, probably, done something wrong. Please, review.

### How to verify it
Run tests

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [x] New test cases have been added and are passing
- [ ] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: tests/Dependency/contracts/OlderDependency.sol [hunk 1]
```diff
@@ -0,0 +1,7 @@
+pragma solidity ^0.4.0;
+
+contract OlderDependency {
+    function foo() pure public returns(bool) {
+        return true;
+    }
+}
```
Comment (unparalleled-js) [inline tests/Dependency/contracts/OlderDependency.sol:7]: Missing newline at EOF

File: tests/contracts/ImportOlderDependency.sol [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+
+import "@remapping/contracts/OlderDependency.sol";
+
+contract ImportOlderDependency {
+    function foo() pure public returns(bool) {
+        return true;
+    }
+}
```
Comment (unparalleled-js) [inline tests/contracts/ImportOlderDependency.sol:9]: Missing newline at EOF

**Review by unparalleled-js — APPROVED** [review]
A clean solution, I like it!
Thank you kindly :)

A couple nits about missing newlines

**Comment by folkyatina** [comment]
Fixed newlines. 

And there is a strange test failing in irrelevant place. Honestly, have no idea if it's connected with PR. It was failing on my machine once I cloned the repo.

**Comment by unparalleled-js** [comment]
> And there is a strange test failing in irrelevant place.

The test failing is not connected with the PR; it is failing across all our plugins and it will be resolved soon!

**Review by unparalleled-js — APPROVED** [review]

**Review by unparalleled-js — DISMISSED** [review]

File: ape_solidity/compiler.py [hunk 3]
```diff
@@ -241,6 +241,11 @@ def _get_compiler_settings(self, version_map: Dict, base_path: Path) -> Dict[Ver
 
             if cli_base_path:
                 settings["base_path"] = cli_base_path
+            else:
+                settings["import_remappings"] = {
+                    i: str(contracts_path / relative_path)
```
Comment (unparalleled-js) [inline ape_solidity/compiler.py:246]:
```suggestion
                    i: str(base_path / relative_path)
```

**Comment by unparalleled-js** [comment]
The new test was added since your work started and your work _does_ actually affect this test.
We need to update the expected value for the remapping settings, if you would!

**Comment by unparalleled-js** [comment]
(note: in the future, can you create PRs using a branch other than main? It makes it easier for me to assist!)

**Review by unparalleled-js — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**